### ISSUE TITLE
`pin_compatible(..., exact=True)` does not work

### ISSUE THREAD

**Issue by borchero** [prefix-dev/rattler-build#1242]
In the latest version of `rattler-build`, using `pin_compatible` with an argument `exact=True` reliably makes the build fail. I would expect that the function would simply pin the version and build string resolved during the build.

NOTE: I can reliably build with `pin_compatible(..., exact=True)` using `rattler-build=0.20.0`.

---

MWE:

```yaml
package:
  name: test
  version: v1

requirements:
  host:
    - python
  run:
    - ${{ pin_compatible('python', exact=True) }}
```

fails with

```
Error:   × Missing output: python (used in pin_subpackage)
```

**Comment by wolfv** [comment]
Whoops, that looks like a proper bug. Will look into it right away.

**Comment by wolfv** [comment]
Fixed by #1122
